src/queryCache.js

```diff
@@ -313,10 +313,12 @@ export function makeQueryCache() {
 
         // Do we need to retry the request?
         if (
-          // Only retry if the document is visible
           query.config.retry === true ||
-          query.state.failureCount <= query.config.retry
+          (typeof query.config.retry == 'number' &&
```

Comment:
I don't think we need this check here. If retry is a function, this condition will be falsy